### PR TITLE
SUS-2052 added php execution timeout reporter

### PR DESCRIPTION
**Pull request by pwojcik86**
https://wikia-inc.atlassian.net/browse/SUS-2052

This PR adds a reporter for PHP execution timeouts. Since this is usually caused by extremely large articles, the tickets should be sent to ComSup for investigation.

@mixth-sense @macbre

### CODE REVIEW

File: reporter/bin/check.py [hunk 1]
```diff
@@ -77,6 +77,9 @@
 reports += ChatLogsSource().query('uncaughtException', threshold=1)
 reports += ChatLogsSource().query('SyntaxError', threshold=1)
 
+# @see https://kibana5.wikia-inc.com/app/kibana#/discover/b4a08460-3633-11e7-8beb-4b869586dfbf?_g=(time%3A(from%3Anow-6h%2Cmode%3Aquick%2Cto%3Anow))
+reports += PHPExecutionTimeoutSource(period=21600).query(threshold=5)
```
Comment (macbre) [inline reporter/bin/check.py]: Why a custom `period` here?

File: reporter/sources/php/execution_timeouts.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class PHPExecutionTimeoutSource(PHPLogsSource):
+    """
+    Source of PHP execution timeouts.
+    Since this problem is commonly caused by extremely large articles,
+    reports should be sent to Community Support.
+    """
+
+    REPORT_LABEL = "PHP Execution Timeout"
```
Comment (macbre) [inline reporter/sources/php/execution_timeouts.py]: Let's create tickets that have this label directly in ComSup project.

The code that can do that - https://github.com/Wikia/jira-reporter/blob/master/reporter/classifier/classifier.py#L57-L70

File: reporter/bin/check.py (hunk 1 above)
Comment (pwojcik86) [inline reporter/bin/check.py]: This is just a guess, but I'm not convinced that these errors will always be closely grouped. Do you think that a longer period will be a problem?

Comment (macbre) [inline reporter/bin/check.py]: Good point. Let's try 6 hours period.

File: reporter/sources/php/execution_timeouts.py (hunk 2 above)
Comment (pwojcik86) [inline reporter/sources/php/execution_timeouts.py]: Done

File: reporter/classifier/classifier.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ class Classifier(object):
     """
     PROJECT_MAIN = 'MAIN'
     PROJECT_SER = 'SER'
+    PROJECT_COMMUNITY_SUPPORT = 'CS'
```
Comment (macbre) [inline reporter/classifier/classifier.py]: :+1:

File: reporter/classifier/classifier.py [hunk 4]
```diff
@@ -69,6 +70,9 @@ def classify(self, report):
         if ChatLogsSource.REPORT_LABEL in labels:
             return self.PROJECT_MAIN, self.get_component_id('Chat')
 
+        if PHPExecutionTimeoutSource.REPORT_LABEL in labels:
+            return self.PROJECT_COMMUNITY_SUPPORT, None
```
Comment (macbre) [inline reporter/classifier/classifier.py]: `component_id` needs to be provided:

```python
ticket_dict['components'] = [{'id': str(component_id)}]  # "expected 'id' property to be a string"
```

Comment (pwojcik86) [inline reporter/classifier/classifier.py]: @timmyquivy said that specifying a component in not necessary. I made a few changes that will allow the component to be optional.

**Pull request merged**